src/test/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperationTest.java

```diff
@@ -308,56 +483,165 @@ void shouldReturnEmptyListIfUserHasAccessToNoCasesForRA() {
+
+            // ASSERT
+            verify(roleAssignmentService).createCaseRoleAssignments(
+                caseDetailsCaptor.capture(),
+                eq(USER_ID),
+                rolesCaptor.capture(),
+                eq(false)
+            );
+
+            CaseDetails caseDetails = caseDetailsCaptor.getValue();
+            assertAll(
+                () -> assertEquals(CASE_ID.toString(), caseDetails.getId()),
+                () -> assertEquals(CASE_REFERENCE, caseDetails.getReference())
+            );
+
+            Set<String> roles = rolesCaptor.getValue();
+            assertAll(
+                () -> assertEquals(1, roles.size()),
+                () -> assertTrue(roles.contains(CASE_ROLE))
+            );
+
+            verifyZeroInteractions(caseUserRepository);
         }
 
         @Test
         @DisplayName("should not add case user role when same role with different case exists")
         void shouldNotAddCaseUserRoleWhenRoleIsCaseInsensitive() {
+
             // ARRANGE
+            when(applicationParams.getEnableAttributeBasedAccessControl()).thenReturn(false);
+
             String role = "[DEFENDANT]";
             List<CaseAssignedUserRoleWithOrganisation> caseUserRoles = Lists.newArrayList(
                 new CaseAssignedUserRoleWithOrganisation(CASE_REFERENCE.toString(), USER_ID, role)
             );
-            given(caseUserRepository.findCaseRoles(CASE_ID, USER_ID)).willReturn(List.of("[defendant]"));
+            // register existing case role
+            mockExistingCaseUserRoles(List.of(
+                createCaseUserEntity(CASE_ID, "[defendant]", USER_ID))
+            );
 
             // ACT
             caseAccessOperation.addCaseUserRoles(caseUserRoles);
 
             // ASSERT
             verify(caseUserRepository, never()).grantAccess(CASE_ID, USER_ID, role);
+
+            verifyZeroInteractions(roleAssignmentService);
+        }
+
+        @Test
+        @DisplayName("RA set to true, should not add case user role when same role with different case exists")
+        void shouldNotAddCaseUserRoleWhenRoleIsCaseInsensitiveForRA() {
+
+            // ARRANGE
+            when(applicationParams.getEnableAttributeBasedAccessControl()).thenReturn(false);
+
+            String role = "[DEFENDANT]";
+            List<CaseAssignedUserRoleWithOrganisation> caseUserRoles = Lists.newArrayList(
+                new CaseAssignedUserRoleWithOrganisation(CASE_REFERENCE.toString(), USER_ID, role)
+            );
+            // register existing case role
+            mockExistingCaseUserRolesForRA(List.of(
+                new CaseAssignedUserRole(CASE_REFERENCE.toString(), USER_ID, "[defendant]")
+            ));
+
+            // ACT
+            caseAccessOperation.addCaseUserRoles(caseUserRoles);
+
+            // ASSERT
+            verify(caseUserRepository, never()).grantAccess(CASE_ID, USER_ID, role);
+
+            verifyZeroInteractions(roleAssignmentService);
```

Comment:
This test needs updating for `getEnableAttributeBasedAccessControl()).thenReturn(true)`

Reply:
Resolved in commit cd1aaacf05d0591bb6b7338046d4c07073b49139.